### PR TITLE
Selenium webdriver v3

### PR DESCRIPTION
**Pull request by SupernaviX**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/SeleniumHQ/selenium/blob/55e370c99a289d36a6ecc41978f7fe2d3813b21c/javascript/node/selenium-webdriver/CHANGES.md
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by SupernaviX** [comment]
@BillArmstrong @kuniwak @cnishina could you give this a look?

**Comment by cnishina** [comment]
I'm on it. @SupernaviX Thanks for the PR! I was going eventually write these. So there are 30 files to review against the selenium-webdriver@3.0.1 release. This is going to take some time, please be patient.

@andy-ms do you know if moving older files to a `v2/` directory good practice? This is a pretty big change. Also since selenium-webdriver changes are not really follow semantic versioning, should we call this directory `v2.53`? When I wrote the 2.53.X types, they were very different from 2.44.X types.

**Comment by SupernaviX** [comment]
Thanks! I'm sure it'll take some time, I just tagged the other contributors because the bot didn't for whatever reason.

As for the major version changes, I was following the guide from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version) which said to do it for major version changes. I'm not sure how different minor versions are supposed to work with DT and the types repository.

**Comment by andy-ms** [comment]
We only support major versions for directories. Making a `v2` folder is good.

**Review by cnishina — CHANGES_REQUESTED** [review]
A few minor changes. I still need to review the index.d.ts.

File: selenium-webdriver/http.d.ts
Comment (SupernaviX) [inline selenium-webdriver/http.d.ts]: https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/http/index.js#L252 It looks like `class Executor` is still exported there, though the actual definition is in another file.

Comment (SupernaviX) [inline selenium-webdriver/http.d.ts]: On closer inspection, HttpRequest and HttpResponse are named wrong (though apparently that's been the case), I'll fix those

File: selenium-webdriver/remote.d.ts
Comment (SupernaviX) [inline selenium-webdriver/remote.d.ts]: Sometimes it's a fun challenge figuring out how to express perfectly valid JS constructs in a static type system.

File: selenium-webdriver/ie.d.ts
Comment (SupernaviX) [inline selenium-webdriver/ie.d.ts]: The implementations of some of the other createSession methods accept either capabilities or a `{desired, required}` hash. Should we match the JSDoc for those or the implementation?

File: selenium-webdriver/http.d.ts
Comment (cnishina) [inline selenium-webdriver/http.d.ts]: Huzzah for pull request reviews! I might get to the index.d.ts maybe on Monday.

File: selenium-webdriver/index.d.ts
Comment (cnishina) [inline selenium-webdriver/index.d.ts]: We should we follow the createSession args here and not pass it `...var_args: any[]`:

```
static createSession(
        executor, capabilities, opt_flow, opt_ctor, opt_onQuit) {
```

**Review by cnishina — CHANGES_REQUESTED** [review]
Please see comments for index.d.ts

**Comment by cnishina** [comment]
Done with the review! Whew.

File: selenium-webdriver/index.d.ts
Comment (SupernaviX) [inline selenium-webdriver/index.d.ts]: See the comment on the line above. The child classes of WebDriver have their own createSession methods with incompatible signatures, which typescript forbids. We can either pass this `...var_args: any[]` or define all arguments as type `any` (which may not work anyway, I'd have to check).

File: selenium-webdriver/index.d.ts
Comment (SupernaviX) [inline selenium-webdriver/index.d.ts]: I assumed these were internal only, but I have no problem with adding them.

File: selenium-webdriver/index.d.ts
Comment (SupernaviX) [inline selenium-webdriver/index.d.ts]: I think this makes their signature actually compatible with native promises!

File: selenium-webdriver/index.d.ts
Comment (SupernaviX) [inline selenium-webdriver/index.d.ts]: On closer inspection, these are static methods on Thenable, which are already defined.

File: selenium-webdriver/index.d.ts
Comment (SupernaviX) [inline selenium-webdriver/index.d.ts]: I just checked, defining this method as

```
static createSession(
        executor: any, capabilities: any, opt_flow?: any, opt_ctor?: any, opt_onQuit?: any) 
```
successfully compiles. Would you prefer that?

**Comment by cnishina** [comment]
Please rebase.

**Comment by SupernaviX** [comment]
rebased

**Comment by jasonruesch** [comment]
Do you know when this will be getting merged?

**Comment by cnishina** [comment]
@andy-ms We should get #14604 merged first so the typings can be published for 2 and then we can merge this one for version 3?

**Comment by minestarks** [comment]
@cnishina merged #14604 , will be good to go after the merge conflict has been resolved here.

**Comment by cnishina** [comment]
@SupernaviX Could we get this rebased?

**Comment by SupernaviX** [comment]
@cnishina @minestarks rebased and v2 typings have been updated

**Pull request merged**
